### PR TITLE
feat: After archiving, close the task modal

### PR DESCRIPTION
**Pull request by juniusfree**
Task: https://app.wonderverse.xyz/dashboard?task=46094456565268510&view=grid

Changes:
1. Feat: close the task view modal after archiving a task
2. Fix: Undoing a task reverts to the initial status instead of the recent status. To reproduce:
    1. View a task then close
    2. Move the card to a new status (e.g. to do -> in progress)
    3. View the task then archive
    4. Undo

**Before the update**

https://user-images.githubusercontent.com/8164667/162601131-0bfaa477-addf-4a9e-a85a-66e411af4960.mp4

**After the update**

https://user-images.githubusercontent.com/8164667/162601138-956dd6b4-283f-4579-95ee-5e69f94274cd.mp4

### CODE REVIEW

File: wondrous-app/components/Common/Task/modal.tsx [hunk 1]
```diff
@@ -773,7 +773,6 @@ export const TaskViewModal = (props: ITaskListModalProps) => {
 
   const [getTaskById] = useLazyQuery(GET_TASK_BY_ID, {
     fetchPolicy: 'network-only',
-    nextFetchPolicy: 'cache-and-network',
```
Comment (andywong418) [inline wondrous-app/components/Common/Task/modal.tsx:776]: Why do we remove this @juniusfree ?

**Review by andywong418 — COMMENTED** [review]
Just a comment on fetch policy change!

File: wondrous-app/components/Common/Task/modal.tsx (hunk 1 above)
Comment (juniusfree) [reply wondrous-app/components/Common/Task/modal.tsx]: @andywong418 I reverted this change. I thought it's contributing to the issue #2 that I mentioned above.

**Comment by andywong418** [comment]
@juniusfree No need to revert! If it's causing and fixing issue 2 then please put back :) Was just wondering

**Comment by juniusfree** [comment]
> @juniusfree No need to revert! If it's causing and fixing issue 2 then please put back :) Was just wondering

@andywong418 It's working even if reverted. I re-tested it. The issue is fixed with the other changes that I made. 😅

**Pull request merged**